test/built-ins/Function/prototype/toString/intrinsics.js

```diff
@@ -0,0 +1,25 @@
+// Copyright (C) 2016 Michael Ficarra. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-function.prototype.tostring
+description: Function.prototype.toString on well-known intrinsic function objects
+---*/
+
+let intrinsics = {
+  Array, ArrayBuffer, Boolean, DataView, Date, decodeURI, decodeURIComponent, encodeURI,
+  encodeURIComponent, Error, eval, EvalError, Float32Array, Float64Array, Function, Int8Array,
+  Int16Array, Int32Array, isFinite, isNaN, Map, Number, Object, parseFloat, parseInt, Promise,
+  Proxy, RangeError, ReferenceError, RegExp, Set, String, Symbol, SyntaxError, TypeError,
+  Uint8Array, Uint8ClampedArray, Uint16Array, Uint32Array, URIError, WeakMap, WeakSet,
+};
+
+const NATIVE_FUNCTION_RE = /\bfunction\b[\s\S]+\b\w+\b[\s\S]*\([\s\S]*\)[\s\S]*\{[\s\S]*\[[\s\S]*\bnative\b[\s\S]+\bcode\b[\s\S]*\][\s\S]*\}/;
```

Comment:
That's fine, I expect invalid strings to pass. The point is that a valid string cannot fail the regex. Remember, comments are allowed in any whitespace position. Without employing a full-blown tokeniser here, I think this is the best we can do.